src/libsystemd/sd-netlink/netlink-message.c

```diff
@@ -1093,38 +1093,39 @@ int sd_netlink_message_read_strv(sd_netlink_message *m, unsigned short container
         return 0;
 }
 
-static int netlink_container_parse(sd_netlink_message *m,
-                                   struct netlink_container *container,
-                                   struct rtattr *rta,
-                                   size_t rt_len) {
+static int netlink_container_parse(
+                sd_netlink_message *m,
+                struct netlink_container *container,
+                struct rtattr *rta,
+                size_t rt_len) {
+
         _cleanup_free_ struct netlink_attribute *attributes = NULL;
-        size_t n = 0;
+        uint16_t max_attr = 0;
 
         /* RTA_OK() macro compares with rta->rt_len, which is unsigned short, and
          * LGTM.com analysis does not like the type difference. Hence, here we
          * introduce an unsigned short variable as a workaround. */
         unsigned short len = rt_len;
         for (; RTA_OK(rta, len); rta = RTA_NEXT(rta, len)) {
-                unsigned short type;
+                uint16_t attr;
 
-                type = RTA_TYPE(rta);
+                attr = RTA_TYPE(rta);
 
-                if (!GREEDY_REALLOC0(attributes, type + 1))
+                if (!GREEDY_REALLOC0(attributes, attr + 1))
```

Comment:
shouldn't this be:

```c
if (!GREEDY_REALLOC0(attributes, MAX(attr, max_attr) + 1))
```

i mean, yes, greedy realloc is not gonna shrink the memory, but it still looks cleaner to always specify the full amount memory we want to use here.

Reply:
hmm, what about overflows here? whta happens if attr is 0xffff? then the +1 here will overflow. We should probably catch that.